src/lib.rs

```diff
@@ -168,13 +208,6 @@ impl Packet {
             None
```

Comment:
Should we return some unknown packet variant or an error?

I prefer the error.

Reply:
Hmm. I think an error is fine. But I would like to do a bit of refactoring in the code(including the next_raw_packet function) these days, and I would prefer it if you included in this pull request only the error handling :)). Is that ok?

Reply:
Updated